src/mca/plm/base/plm_base_jobid.c

```diff
@@ -75,24 +75,40 @@ int prrte_plm_base_set_hnp_name(void)
 /*
  * Create a jobid
  */
+static bool reuse = false;
+
 int prrte_plm_base_create_jobid(prrte_job_t *jdata)
 {
+    int16_t i;
+    prrte_jobid_t pjid;
+    prrte_job_t *ptr;
+
     if (PRRTE_FLAG_TEST(jdata, PRRTE_JOB_FLAG_RESTART)) {
         /* this job is being restarted - do not assign it
          * a new jobid
          */
         return PRRTE_SUCCESS;
     }
 
-    if (UINT16_MAX == prrte_plm_globals.next_jobid) {
-        /* if we get here, then no local jobids are available */
-        PRRTE_ERROR_LOG(PRRTE_ERR_OUT_OF_RESOURCE);
-        jdata->jobid = PRRTE_JOBID_INVALID;
-        return PRRTE_ERR_OUT_OF_RESOURCE;
+    if (reuse) {
+        /* find the first unused jobid */
+        for (i=1; i < INT16_MAX; i++) {
+            ptr = NULL;
+            pjid = PRRTE_CONSTRUCT_LOCAL_JOBID(PRRTE_PROC_MY_NAME->jobid, i);
+            prrte_hash_table_get_value_uint32(prrte_job_data, pjid, (void**)&ptr);
+            if (NULL == ptr) {
+                prrte_plm_globals.next_jobid = i;
+                break;
+            }
+        }
     }
 
     /* take the next jobid */
     jdata->jobid =  PRRTE_CONSTRUCT_LOCAL_JOBID(PRRTE_PROC_MY_NAME->jobid, prrte_plm_globals.next_jobid);
     prrte_plm_globals.next_jobid++;
+    if (INT16_MAX == prrte_plm_globals.next_jobid) {
```

Comment:
Can we add a silly check here for:
```
if (reuse && INT16_MAX == prrte_plm_globals.next_jobid) {
   prrte_output(0, "Whoa! What are you doing starting that many jobs concurrently.");
}
else if(INT16_MAX == prrte_plm_globals.next_jobid) {
```

It should never happen, but at least we have a check for it.

Reply:
Then with some kind of error back to the user would be good. Rejecting the spawn call.

Reply:
Added something like that...

Reply:
Perfect. Thanks!